Zend/zend_inheritance.c

```diff
@@ -1146,21 +1162,16 @@ static zend_always_inline inheritance_status do_inheritance_check_on_method_ex(
 		parent = proto;
 	}
 
-	if (!check_only && child->common.prototype != proto && child_zv) {
-		do {
-			if (child->common.scope != ce && child->type == ZEND_USER_FUNCTION) {
-				if (ce->ce_flags & ZEND_ACC_INTERFACE) {
-					/* Few parent interfaces contain the same method */
-					break;
-				} else {
-					/* op_array wasn't duplicated yet */
-					zend_function *new_function = zend_arena_alloc(&CG(arena), sizeof(zend_op_array));
-					memcpy(new_function, child, sizeof(zend_op_array));
-					Z_PTR_P(child_zv) = child = new_function;
-				}
-			}
-			child->common.prototype = proto;
-		} while (0);
+	if (!check_only
+	 && child->common.prototype != proto
+	 /* We are not setting the prototype of overridden interface methods because of abstract
+	  * constructors. See Zend/tests/interface_constructor_prototype_001.phpt. */
+	 && !(ce->ce_flags & ZEND_ACC_INTERFACE)
+	 /* Parent is a trait when its method is abstract. We only need to verify its signature then,
+	  * don't actually use it as a prototype. See Zend/tests/gh14009_004.phpt. */
+	 && !(parent->common.scope->ce_flags & ZEND_ACC_TRAIT)) {
```

Comment:
Right, that was my first thought. But I then realized that this completely disables the error messages, and adding yet another parameter to the already confusing ones is something I was trying to avoid.